src/compiler/transformers/module/module.ts

```diff
@@ -1255,11 +1282,34 @@ namespace ts {
                         ),
                         /*location*/ node.name
                     ),
-                    visitNode(node.initializer, moduleExpressionElementVisitor)
+                    visitNode(node.initializer, nestedElementVisitor)
                 );
             }
         }
 
+        function visitForInOrOfStatement(node: ForInOrOfStatement) {
+            node = visitEachChild(node, nestedElementVisitor, context);
+            if (node.initializer.kind === SyntaxKind.VariableDeclarationList || !destructuringNeedsFlattening(node.initializer)) {
+                return node;
+            }
+            let initializer: ForInitializer = node.initializer;
+            let expression: Expression;
+            if (isIdentifier(node.initializer)) {

```

Comment:
do I need to add the same checks as in `visitUnaryExpression`, i.e. `!isGeneratedIdentifier(...)` and so on?